File: ruby/hamming/hamming.rb
class Hamming
  VERSION = File.read('.version').to_i

  def Hamming.compute(a, b)
    unless a.size == b.size
      raise ArgumentError.new("strands must be the same length")
    end

    a.chars.zip(b.chars).count {|(x, y)| x != y}
  end
end

File: julia/hamming/hamming.jl
function distance(s1::AbstractString, s2::AbstractString)
  if endof(s1) != endof(s2)
    throw(ArgumentError("Strands not the same length."))
  end

  count(zip(s1, s2)) do tuple
    a, b = tuple
    a != b
  end
end
